### PR TITLE
Add feature flag to control service answers feature

### PR DESCRIPTION
**Pull request by reupen**
### Description of change

This is a precursor to #1742 so that we can add new services via data migrations without them appearing in the front end (until the service questions and answers functionality is built).

A corresponding `requires_service_answers_flow_feature_flag` flag has been added to services to indicate which services should be hidden when the feature flag is inactive.

The feature flag will also be used in the next PR to control whether `interaction_questions` is populated for each item in the `/metadata/service/` response.

### Migration SQL

The migration runs the following SQL statements:

```sql
BEGIN;
--
-- Add field requires_service_answers_flow_feature_flag to service
--
ALTER TABLE "metadata_service" ADD COLUMN "requires_service_answers_flow_feature_flag" boolean DEFAULT false NOT NULL;
ALTER TABLE "metadata_service" ALTER COLUMN "requires_service_answers_flow_feature_flag" DROP DEFAULT;
COMMIT;
```

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/metadata/metadata.py [hunk 1]
```diff
@@ -45,6 +49,11 @@
     filterset_class=ServiceFilterSet,
     metadata_id='service',
     model=models.Service,
+    queryset=models.Service.objects.annotate(
```
Comment (currycoder) [inline datahub/metadata/metadata.py]: I feel like this is a fiddly solution.  I appreciate that we are sort of bound to do this based on the way that metadata registration works, but I just want to discuss some possible alternatives:
1) Avoid using the registry at all and create a custom metadata view for this object.  I don't have all of the context for what a registry is used for, though.  Also appreciate that the feature flag will go away at some point so we will end up with a standard registry call anyway.
2) Extend `MetadataMapping` with a `get_queryset` attribute, which is default `None` in the args for `Registry.register()`.  Also extend `datahub.metadata.views._create_metadata_view()` with some code that attaches this as a method to the created class.

The benefit to option 2 IMO is just to have something that is more immediately readable.  Tying in to my other comment though, I'd be happy if this overall flag pattern for metadata was documented somewhere.

**Comment by reupen** [comment]
> I don't understand how the default will be dropped based on the contents of the migration file.

That's the standard Django behaviour – it only uses server-side defaults when adding a column to an existing table, not for row inserts in general.

**Comment by currycoder** [comment]
> That's the standard Django behaviour – it only uses server-side defaults when adding a column to an existing table, not for row inserts in general.

Gotcha - thanks for explaining.  I take it that the table is small enough that we do not need to worry about the multi-step process for adding a default field?

File: datahub/interaction/constants.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Feature flag that controls whether services with requires_service_answers_flow_feature_flag
+# set to True are returned from the API
+SERVICE_ANSWERS_FEATURE_FLAG = 'interaction_service_answers_flow'
```
[the remark this answers is not in the document]
Comment (reupen) [reply datahub/interaction/constants.py]: I think in `constants` has been usual when it's shared between modules. The import interactions one looks like it was in the wrong place (though it's gone now). `__init__.py` is not a good place IMO. Looks like there is still one defined in `datahub/investment/project/proposition/constants.py` which probably should be removed.

File: datahub/feature_flag/utils.py [hunk 3]
```diff
@@ -14,6 +15,13 @@ def is_feature_flag_active(code):
     return FeatureFlag.objects.filter(code=code, is_active=True).exists()
 
 
+def build_is_feature_flag_active_subquery(code):
+    """Return a subquery that checks if a feature flag is active."""
```
[the remark this answers is not in the document]
Comment (reupen) [reply datahub/feature_flag/utils.py]: I'll add an example :+1:

**Comment by reupen** [comment]
Yes, it should be very quick here.

File: datahub/interaction/constants.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (currycoder) [reply datahub/interaction/constants.py]: Cool, I'm happy to use `constants.py` going forward :+1:

File: datahub/metadata/metadata.py (hunk 1 above)
Comment (reupen) [reply datahub/metadata/metadata.py]: I've added some additional comments which hopefully improves things. I think not using the registry loses consistency and I would mainly expect `get_queryset` to be overriden on a view if there was some dynamic logic (e.g. different filtering based on a query parameter or the user).

This module is a little bit cluttered as a good chunk of the metadata models should be in the relevant apps (as they are not shared between apps e.g. `InvestmentProjectStage` should be in the investment project app).

Comment (currycoder) [reply datahub/metadata/metadata.py]: Neat - the improved comments look good.  I agree with what you're saying about consistency.

**Pull request merged**